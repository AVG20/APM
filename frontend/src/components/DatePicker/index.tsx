import generatePicker from 'antd/es/date-picker/generatePicker';
import dayjsGenerateConfig from 'antd/node_modules/rc-picker/lib/generate/dayjs';

const DatePicker = generatePicker(dayjsGenerateConfig);

export default DatePicker;
