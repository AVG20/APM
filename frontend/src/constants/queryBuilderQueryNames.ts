export const COMPOSITE_QUERY = 'compositeQuery';
export const PANEL_TYPES_QUERY = 'panelTypes';
export const PAGE_SIZE = 'pageSize';
