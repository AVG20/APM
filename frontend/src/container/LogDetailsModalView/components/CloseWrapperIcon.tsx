import { CloseOutlined, EditFilled } from '@ant-design/icons';
import { MouseEventHandler } from 'react';

import ICloseWrapperIcon from '../interfaces/ICloseWrapperIcon';

function CloseWrapperIcon({ toggleInput }: ICloseWrapperIcon): JSX.Element {
	const handleClick: MouseEventHandler<Element> = (event) => {
		event.preventDefault();
		event.stopPropagation();
		toggleInput();
	};

	return (
		<div style={{ display: 'flex', gap: 20, marginLeft: -33 }}>
			<span role="button" aria-hidden="true" onClick={handleClick}>
				<EditFilled />
			</span>
			<CloseOutlined />
		</div>
	);
}

export default CloseWrapperIcon;
