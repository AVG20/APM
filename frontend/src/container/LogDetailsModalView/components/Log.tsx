import { Tooltip, Typography } from 'antd';
import dayjs from 'dayjs';
import { useMemo } from 'react';

import ILog from '../interfaces/ILog';
import { LogContainer, LogDiv } from '../styles/Log';

const { Text } = Typography;

function Log({ log }: ILog): JSX.Element {
	const text = useMemo(
		() => `${dayjs(log.timestamp / 1e6).format()} | ${log.body}`,
		[log],
	);

	return (
		<LogContainer>
			<Tooltip title={text}>
				<LogDiv>
					<Text ellipsis>{text}</Text>
				</LogDiv>
			</Tooltip>
		</LogContainer>
	);
}

export default Log;
