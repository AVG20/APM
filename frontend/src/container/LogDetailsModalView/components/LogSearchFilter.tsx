import { Input, InputRef, Popover } from 'antd';
import SearchFields from 'container/LogsSearchFilter/SearchFields';
import { useCallback, useRef, useState } from 'react';

import { Container, DropDownContainer } from '../../LogsSearchFilter/styles';

interface ILogSearchFilter {
	query: string;
	setQuery: (value: string) => void;
}

function LogSearchFilter({ query, setQuery }: ILogSearchFilter): JSX.Element {
	// const [searchText, setSearchText] = useState('');
	const [showDropDown, setShowDropDown] = useState(false);
	const searchRef = useRef<InputRef>(null);

	const onDropDownToggleHandler = useCallback(
		(value: boolean) => (): void => {
			setShowDropDown(value);
		},
		[],
	);

	const onPopOverChange = useCallback(
		(isVisible: boolean) => {
			onDropDownToggleHandler(isVisible)();
		},
		[onDropDownToggleHandler],
	);

	return (
		<Container>
			<Popover
				placement="bottom"
				content={
					<DropDownContainer>
						<SearchFields
							updateQueryString={setQuery}
							onDropDownToggleHandler={onDropDownToggleHandler}
						/>
					</DropDownContainer>
				}
				trigger="click"
				overlayInnerStyle={{
					minWidth: 800,
					width: `${searchRef?.current?.input?.offsetWidth || 0}px`,
				}}
				open={showDropDown}
				destroyTooltipOnHide
				onOpenChange={onPopOverChange}
			>
				<Input.Search
					ref={searchRef}
					placeholder="Search Filter"
					value={query}
					onChange={(e): void => {
						const { value } = e.target;
						setQuery(value);
					}}
					onSearch={setQuery}
					allowClear
				/>
			</Popover>
		</Container>
	);
}

export default LogSearchFilter;
