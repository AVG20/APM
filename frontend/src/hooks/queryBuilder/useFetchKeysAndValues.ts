/* eslint-disable sonarjs/cognitive-complexity */
import { getAttributesValues } from 'api/queryBuilder/getAttributesValues';
import { DEBOUNCE_DELAY } from 'constants/queryBuilderFilterConfig';
import {
	getRemovePrefixFromKey,
	getTagToken,
	isInNInOperator,
} from 'container/QueryBuilder/filters/QueryBuilderSearch/utils';
import useDebounceValue from 'hooks/useDebounce';
import { isEqual, uniqWith } from 'lodash-es';
import { useCallback, useMemo, useRef, useState } from 'react';
import { useDebounce } from 'react-use';
import {
	BaseAutocompleteData,
	DataTypes,
} from 'types/api/queryBuilder/queryAutocompleteResponse';
import { IBuilderQuery } from 'types/api/queryBuilder/queryBuilderData';
import { DataSource } from 'types/common/queryBuilder';

import { useGetAggregateKeys } from './useGetAggregateKeys';

type IuseFetchKeysAndValues = {
	keys: BaseAutocompleteData[];
	results: string[];
	isFetching: boolean;
	sourceKeys: BaseAutocompleteData[];
	handleRemoveSourceKey: (newSourceKey: string) => void;
};

/**
 * Custom hook to fetch attribute keys and values from an API
 * @param searchValue - the search query value
 * @param query - an object containing data for the query
 * @returns an object containing the fetched attribute keys, results, and the status of the fetch
 */

export const useFetchKeysAndValues = (
	searchValue: string,
	query: IBuilderQuery,
	searchKey: string,
): IuseFetchKeysAndValues => {
	const [sourceKeys, setSourceKeys] = useState<BaseAutocompleteData[]>([]);
	const [results, setResults] = useState<string[]>([]);

	const memoizedSearchParams = useMemo(
		() => [
			searchKey,
			query.dataSource,
			query.aggregateOperator,
			query.aggregateAttribute.key,
		],
		[
			searchKey,
			query.dataSource,
			query.aggregateOperator,
			query.aggregateAttribute.key,
		],
	);

	const searchParams = useDebounceValue(memoizedSearchParams, DEBOUNCE_DELAY);

	const isQueryEnabled =
		query.dataSource === DataSource.METRICS
			? !!query.aggregateOperator &&
			  !!query.dataSource &&
			  !!query.aggregateAttribute.dataType
			: true;

	const { isFetching } = useGetAggregateKeys(
		{
			searchText: searchKey,
			dataSource: query.dataSource,
			aggregateOperator: query.aggregateOperator,
			aggregateAttribute: query.aggregateAttribute.key,
			tagType: query.aggregateAttribute.type ?? null,
		},
		{
			queryKey: [searchParams],
			enabled: isQueryEnabled,
			onSuccess: (data) => {
				if (data && data.payload) {
					setSourceKeys((prevState) =>
						uniqWith([...(data.payload.attributeKeys ?? []), ...prevState], isEqual),
					);
				}
			},
		},
	);

	/**
	 * Fetches the options to be displayed based on the selected value
	 * @param value - the selected value
	 * @param query - an object containing data for the query
	 */
	const handleFetchOption = async (
		value: string,
		query: IBuilderQuery,
		keys: BaseAutocompleteData[],
	): Promise<void> => {
		if (!value) {
			return;
		}
		const { tagKey, tagOperator, tagValue } = getTagToken(value);
		const filterAttributeKey = keys.find(
			(item) => item.key === getRemovePrefixFromKey(tagKey),
		);
		setResults([]);

		if (!tagKey || !tagOperator) {
			return;
		}

		const { payload } = await getAttributesValues({
			aggregateOperator: query.aggregateOperator,
			dataSource: query.dataSource,
			aggregateAttribute: query.aggregateAttribute.key,
			attributeKey: filterAttributeKey?.key ?? tagKey,
			filterAttributeKeyDataType: filterAttributeKey?.dataType ?? DataTypes.EMPTY,
			tagType: filterAttributeKey?.type ?? '',
			searchText: isInNInOperator(tagOperator)
				? tagValue[tagValue.length - 1]?.toString() ?? '' // last element of tagvalue will be always user search value
				: tagValue?.toString() ?? '',
		});

		if (payload) {
			const values = Object.values(payload).find((el) => !!el) || [];
			setResults(values);
		}
	};

	const handleRemoveSourceKey = useCallback((sourceKey: string) => {
		setSourceKeys((prevState) =>
			prevState.filter((item) => item.key !== sourceKey),
		);
	}, []);

	// creates a ref to the fetch function so that it doesn't change on every render
	const clearFetcher = useRef(handleFetchOption).current;

	// debounces the fetch function to avoid excessive API calls
	useDebounce(() => clearFetcher(searchValue, query, sourceKeys), 750, [
		clearFetcher,
		searchValue,
		query,
		searchKey,
	]);

	return {
		keys: sourceKeys,
		results,
		isFetching,
		sourceKeys,
		handleRemoveSourceKey,
	};
};
