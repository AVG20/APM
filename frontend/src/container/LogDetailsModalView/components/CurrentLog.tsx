import { Tooltip, Typography } from 'antd';
import dayjs from 'dayjs';
import { useMemo } from 'react';

import { ICurrentLog } from '../interfaces/ILog';
import { LogContainer, LogDiv } from '../styles/Log';

const { Text } = Typography;

function CurrentLog({ log }: ICurrentLog): JSX.Element {
	const text = useMemo(
		() => `${dayjs(log.timestamp / 1e6).format()} | ${log.body}`,
		[log],
	);

	return (
		<LogContainer>
			<LogDiv>
				<Tooltip title={text}>
					<Text ellipsis>{text}</Text>
				</Tooltip>
			</LogDiv>
		</LogContainer>
	);
}

export default CurrentLog;
