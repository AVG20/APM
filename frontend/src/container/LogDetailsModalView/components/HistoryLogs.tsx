import { Button, Spin, Typography } from 'antd';

import IHistoryLogs, { HistoryPosition } from '../interfaces/IHistoryLogs';
import {
	HisoryLogContainer,
	HistoryLogHeader,
	HistoryLogsScrolled,
	LogContainer,
	LogDiv,
	SpinnerContainer,
} from '../styles/Log';
import Log from './Log';

const { Text } = Typography;

function HistoryLogs({
	logs,
	position,
	addMoreLogs,
	isLoad,
	isError,
}: IHistoryLogs): JSX.Element {
	if (isError) {
		return <div>Something went wrong</div>;
	}

	return (
		<>
			{position === HistoryPosition.prev && (
				<HistoryLogHeader>
					Showing 10 lines after match
					<Button size="small" onClick={addMoreLogs}>
						Show 10 more lines
					</Button>
				</HistoryLogHeader>
			)}
			<HisoryLogContainer>
				{isLoad ? (
					<SpinnerContainer>
						<Spin />
					</SpinnerContainer>
				) : (
					<HistoryLogsScrolled>
						{logs?.length === 0 ? (
							<LogContainer>
								<LogDiv>
									<Text ellipsis>There are no logs</Text>
								</LogDiv>
							</LogContainer>
						) : (
							logs?.map((log) => (
								<Log key={log.data.id} timestamp={log.timestamp} log={log} />
							))
						)}
					</HistoryLogsScrolled>
				)}
			</HisoryLogContainer>
			{position === HistoryPosition.next && (
				<HistoryLogHeader>
					Showing 10 lines before match
					<Button size="small" onClick={addMoreLogs}>
						Show 10 more lines
					</Button>
				</HistoryLogHeader>
			)}
		</>
	);
}

export default HistoryLogs;
