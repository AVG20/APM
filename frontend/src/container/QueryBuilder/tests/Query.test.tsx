import { render } from '@testing-library/react';
import { PANEL_TYPES } from 'constants/queryBuilder';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import i18n from 'ReactI18';
import store from 'store';

import { Query } from '../components';
import { queryMockData } from '../mock/queryData';

describe('QueryBuilder: Main Query wrapper', () => {
	const queryClient = new QueryClient({
		defaultOptions: {
			queries: {
				refetchOnWindowFocus: false,
			},
		},
	});

	it('it should render Query', () => {
		const { asFragment } = render(
			<MemoryRouter>
				<Provider store={store}>
					<I18nextProvider i18n={i18n}>
						<QueryClientProvider client={queryClient}>
							<Query
								index={0}
								isAvailableToDisable={false}
								queryVariant="static"
								query={queryMockData}
								panelType={PANEL_TYPES.TIME_SERIES}
							/>
						</QueryClientProvider>
					</I18nextProvider>
				</Provider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
