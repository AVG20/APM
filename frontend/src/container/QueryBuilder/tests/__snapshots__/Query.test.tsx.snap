// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryBuilder: Main Query wrapper it should render Query 1`] = `
<DocumentFragment>
  .c7 {
  margin-right: 0.6875rem;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c6 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0.875rem;
  min-height: 1.375rem;
  cursor: pointer;
}

.c8 {
  pointer-events: none;
}

.c3 {
  padding: 0 0.6875rem;
  min-height: 2rem;
  width: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.125rem;
  border: 0.0625rem solid #434343;
  background-color: #141414;
}

.c1 {
  position: absolute;
  top: 0.5rem;
  right: 0.9375rem;
  z-index: 1;
  cursor: pointer;
  opacity: 0.45;
  width: 1.3125rem;
  height: 1.3125rem;
}

.c1 svg {
  width: 100%;
  height: 100%;
}

.c0 {
  padding-right: 3rem;
}

.c2 {
  min-width: 2rem;
  height: 2.25rem;
  padding: 0.43rem 0.68rem;
  border-radius: 0.375rem;
  margin-right: 0.5rem;
  pointer-events: none;
}

.c4 {
  width: auto;
  cursor: pointer;
  pointer-events: auto;
}

.c5 {
  width: 10rem;
  cursor: pointer;
  pointer-events: auto;
}

<div
    class="ant-row c0 css-dev-only-do-not-override-1i536d8"
    style="margin-top: -7.5px; margin-bottom: -7.5px;"
  >
    <span
      aria-label="close-circle"
      class="anticon anticon-close-circle c1"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="close-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
        />
        <path
          d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
        />
      </svg>
    </span>
    <div
      class="ant-col ant-col-24 css-dev-only-do-not-override-1i536d8"
      style="padding-top: 7.5px; padding-bottom: 7.5px;"
    >
      <div
        class="ant-row ant-row-middle css-dev-only-do-not-override-1i536d8"
      >
        <div
          class="ant-col css-dev-only-do-not-override-1i536d8"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-1i536d8 ant-btn-primary c2"
            type="button"
          >
            <span>
              A
            </span>
          </button>
          <div
            class="c3"
          >
            Metrics
          </div>
        </div>
        <div
          class="ant-col css-dev-only-do-not-override-1i536d8"
          style="flex: 1;"
        >
          <div
            class="ant-row css-dev-only-do-not-override-1i536d8"
            style="margin: -2.5px -5.5px -2.5px -5.5px;"
          >
            <div
              class="ant-col css-dev-only-do-not-override-1i536d8"
              style="padding: 2.5px 5.5px 2.5px 5.5px;"
            >
              <div
                class="c3"
              >
                WHERE
              </div>
            </div>
            <div
              class="ant-col css-dev-only-do-not-override-1i536d8"
              style="padding: 2.5px 5.5px 2.5px 5.5px; flex: 1;"
            >
              <div
                class="ant-select css-dev-only-do-not-override-1i536d8 ant-select-multiple ant-select-show-search"
                style="width: 100%; min-width: 10rem;"
              >
                <div
                  class="ant-select-selector"
                >
                  <div
                    class="ant-select-selection-overflow"
                  >
                    <div
                      class="ant-select-selection-overflow-item"
                      style="opacity: 1;"
                    >
                      <span>
                        <span
                          class="ant-tag css-dev-only-do-not-override-1i536d8"
                        >
                          <span
                            class="ant-typography ant-typography-ellipsis ant-typography-single-line ant-typography-ellipsis-single-line c4 css-dev-only-do-not-override-1i536d8"
                          >
                            resource_signoz_collector_id = 1a5d3cc2-4b3e-4c7c-ad07-c4cdd739d1b9
                          </span>
                          <span
                            aria-label="close"
                            class="anticon anticon-close ant-tag-close-icon"
                            role="img"
                            tabindex="-1"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="close"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                              />
                            </svg>
                          </span>
                        </span>
                      </span>
                    </div>
                    <div
                      class="ant-select-selection-overflow-item"
                      style="opacity: 1;"
                    >
                      <span>
                        <span
                          class="ant-tag css-dev-only-do-not-override-1i536d8"
                        >
                          <span
                            class="ant-typography ant-typography-ellipsis ant-typography-single-line ant-typography-ellipsis-single-line c5 css-dev-only-do-not-override-1i536d8"
                          >
                            service_name IN frontend,operation,ser
                          </span>
                          <span
                            aria-label="close"
                            class="anticon anticon-close ant-tag-close-icon"
                            role="img"
                            tabindex="-1"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="close"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                              />
                            </svg>
                          </span>
                        </span>
                      </span>
                    </div>
                    <div
                      class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
                      style="opacity: 1;"
                    >
                      <div
                        class="ant-select-selection-search"
                        style="width: 0px;"
                      >
                        <input
                          aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                          aria-autocomplete="list"
                          aria-controls="rc_select_TEST_OR_SSR_list"
                          aria-haspopup="listbox"
                          aria-owns="rc_select_TEST_OR_SSR_list"
                          autocomplete="off"
                          class="ant-select-selection-search-input"
                          id="rc_select_TEST_OR_SSR"
                          role="combobox"
                          type="search"
                          value=""
                        />
                        <span
                          aria-hidden="true"
                          class="ant-select-selection-search-mirror"
                        >
                          Â 
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-11 css-dev-only-do-not-override-1i536d8"
      style="padding-top: 7.5px; padding-bottom: 7.5px;"
    >
      <div
        class="ant-row css-dev-only-do-not-override-1i536d8"
        style="margin: -2.5px -5.5px -2.5px -5.5px;"
      >
        <div
          class="ant-col css-dev-only-do-not-override-1i536d8"
          style="padding: 2.5px 5.5px 2.5px 5.5px; flex: 0 0 5.93rem;"
        >
          <div
            class="ant-select css-dev-only-do-not-override-1i536d8 ant-select-single ant-select-show-arrow ant-select-show-search"
            style="width: 100%; min-width: 10rem;"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  role="combobox"
                  type="search"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-item"
                title="Noop"
              >
                Noop
              </span>
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
        <div
          class="ant-col css-dev-only-do-not-override-1i536d8"
          style="padding: 2.5px 5.5px 2.5px 5.5px; flex: 1 1 12.5rem;"
        >
          <div
            class="ant-select ant-select-auto-complete css-dev-only-do-not-override-1i536d8 ant-select-single ant-select-show-search"
            style="width: 100%; min-width: 10rem;"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  role="combobox"
                  type="search"
                  value="signoz_latency_count"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-11 ant-col-offset-2 css-dev-only-do-not-override-1i536d8"
      style="padding-top: 7.5px; padding-bottom: 7.5px;"
    >
      <div
        class="ant-row css-dev-only-do-not-override-1i536d8"
        style="margin: -2.5px -5.5px -2.5px -5.5px;"
      >
        <div
          class="ant-col css-dev-only-do-not-override-1i536d8"
          style="padding: 2.5px 5.5px 2.5px 5.5px; flex: 0 0 5.93rem;"
        >
          <div
            class="c3"
          >
            Group by
          </div>
        </div>
        <div
          class="ant-col css-dev-only-do-not-override-1i536d8"
          style="padding: 2.5px 5.5px 2.5px 5.5px; flex: 1 1 12.5rem;"
        >
          <div
            class="ant-select css-dev-only-do-not-override-1i536d8 ant-select-multiple ant-select-show-search"
            style="width: 100%; min-width: 10rem;"
          >
            <div
              class="ant-select-selector"
            >
              <div
                class="ant-select-selection-overflow"
              >
                <div
                  class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
                  style="opacity: 1;"
                >
                  <div
                    class="ant-select-selection-search"
                    style="width: 0px;"
                  >
                    <input
                      aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                      aria-autocomplete="list"
                      aria-controls="rc_select_TEST_OR_SSR_list"
                      aria-haspopup="listbox"
                      aria-owns="rc_select_TEST_OR_SSR_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="rc_select_TEST_OR_SSR"
                      role="combobox"
                      type="search"
                      value=""
                    />
                    <span
                      aria-hidden="true"
                      class="ant-select-selection-search-mirror"
                    >
                      Â 
                    </span>
                  </div>
                </div>
              </div>
              <span
                class="ant-select-selection-placeholder"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-24 css-dev-only-do-not-override-1i536d8"
      style="padding-top: 7.5px; padding-bottom: 7.5px;"
    >
      <div
        class="ant-row css-dev-only-do-not-override-1i536d8"
      >
        <div
          class="ant-col ant-col-24 css-dev-only-do-not-override-1i536d8"
        >
          <div
            class="ant-col c6 css-dev-only-do-not-override-1i536d8"
          >
            <span
              aria-label="plus-square"
              class="anticon anticon-plus-square c7"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="plus-square"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M328 544h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
                />
                <path
                  d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
                />
              </svg>
            </span>
            <span>
              Add conditions for 
              <span>
                <a
                  class="ant-typography c8 css-dev-only-do-not-override-1i536d8"
                >
                  AGGREGATION INTERVAL
                </a>
                 
              </span>
              and 
              <a
                class="ant-typography c8 css-dev-only-do-not-override-1i536d8"
              >
                HAVING
              </a>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row css-dev-only-do-not-override-1i536d8"
      style="width: 100%;"
    >
      <span
        class="ant-input-group-wrapper css-dev-only-do-not-override-1i536d8"
      >
        <span
          class="ant-input-wrapper ant-input-group css-dev-only-do-not-override-1i536d8"
        >
          <span
            class="ant-input-group-addon"
          >
            Legend Format
          </span>
          <input
            class="ant-input css-dev-only-do-not-override-1i536d8"
            type="text"
            value=""
          />
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
